scertec.go

```diff
@@ -99,10 +105,20 @@ func (c *Client) GetCertificate(hello *tls.ClientHelloInfo) (*tls.Certificate, e
 		typ = "ecdsa"
 	}
 	secName := secretName(c.prefix, hello.ServerName, typ)
+
 	sec := c.st.Secret(secName)
+
+	// In the common case the above suceeds, once setec has seen it the first time.
+	// But if it's not there, we'll look it up with a timeout.
 	if sec == nil {
-		return nil, errors.New("invalid server name")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
FWIW the lookups already have a timeout baked in if you don't set one (though it'll prefer yours if it's there). The baked-in one happens to be 5m already, though 🙂

Reply:
> 5m already

This is 5 _seconds_ if you misread the two fives as being the same unit.

Reply:
I did 😬